client/service/service.go

```diff
@@ -56,27 +57,78 @@ func New(config, dataDir string, platIfce libbox.PlatformInterface) (*BoxService
 	if err := libbox.Setup(setupOpts); err != nil {
 		return nil, fmt.Errorf("setup libbox: %w", err)
 	}
-	lb, err := libbox.NewServiceWithContext(ctx, config, platIfce)
-	if err != nil {
-		return nil, fmt.Errorf("create libbox service: %w", err)
+
+	return bs, nil
+}
+
+func (bs *BoxService) Start() error {
+	bs.mu.Lock()
+	defer bs.mu.Unlock()
+
+	if bs.isRunning.Load() {
+		return errors.New("service is already running")
 	}
 
-	bs := &BoxService{
-		libbox:       lb,
-		ctx:          ctx,
-		pauseManager: service.FromContext[pause.Manager](ctx),
-		pauseAccess:  sync.Mutex{},
+	// (re)-initialize the libbox service
+	lb, ctx, err := newLibboxService(bs.config, bs.platIfce)
+	if err != nil {
+		return err
 	}
 
-	return bs, nil
+	bs.libbox = lb
+	bs.ctx = ctx
+	bs.pauseManager = service.FromContext[pause.Manager](ctx)
+
+	if err = lb.Start(); err == nil {
+		bs.isRunning.Store(true)
+	}
+	return err
 }
 
-func (bs *BoxService) Start() error {
-	return bs.libbox.Start()
+// newLibboxService creates a new libbox service with the given config and platform interface
+func newLibboxService(config string, platIfce libbox.PlatformInterface) (*libbox.BoxService, context.Context, error) {
+	// Retrieve protocol registries
+	inboundRegistry, outboundRegistry, endpointRegistry := protocol.GetRegistries()
+	ctx := box.Context(
+		context.Background(),
+		inboundRegistry,
+		outboundRegistry,
+		endpointRegistry,
+	)
+
+	// initialize the libbox service
+	lb, err := libbox.NewServiceWithContext(ctx, config, platIfce)
+	if err != nil {
+		return nil, nil, fmt.Errorf("create libbox service: %v", err)
+	}
+
+	return lb, ctx, nil
 }
 
 func (bs *BoxService) Close() error {
-	return bs.libbox.Close()
+	bs.mu.Lock()
+	defer bs.mu.Unlock()
+
+	if !bs.isRunning.Load() {
+		return errors.New("service already stopped")
+	}
+
+	// Clear pause timer
+	if bs.pauseTimer != nil {
+		bs.pauseTimer.Stop()
+		bs.pauseTimer = nil
+	}
+
+	if bs.libbox != nil {
+		err := bs.libbox.Close()
```

Comment:
After we call `Close()` here, I think the BoxService instance is no longer usable, and that's why we're getting "file already closed" errors. The changes above make sure to rebuild it before restarting